cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -163,99 +150,308 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
+        slices.reserve(arr.ndim());
+        for (size_t i = 0; i < arr.ndim(); ++i)
+        {
+            small_vector<int> default_slice(3);
+            default_slice[0] = 0; // start
+            default_slice[1] = arr.shape(i); // stop
+            default_slice[2] = 1; // step
+            slices.push_back(std::move(default_slice));
+        }
+        return slices;
+    }
+
+    static void process_slice(pybind11::tuple const & tuple,
+                              std::vector<small_vector<int>> & slices,
+                              size_t ndim)
+    {
+        namespace py = pybind11;
+
+        // copy slices from the front untill an ellipsis
+        bool ellipsis_flag = false;
+        for (auto it = tuple.begin(); it != tuple.end(); it++)
+        {
+            if (py::isinstance<py::ellipsis>(*it))
+            {
+                // stop here and iterator the tuple from back later
+                ellipsis_flag = true;
+                break;
+            }
+
+            auto & slice_out = slices[it - tuple.begin()];
+            const auto slice_in = (*it).cast<py::slice>();
+
+            copy_slice(slice_out, slice_in);
+        }
+
+        // copy slices from the back until an ellipsis
+        if (ellipsis_flag)
+        {
+            for (size_t size = 0; size < tuple.size(); size++)
+            {
+                auto it = tuple.end() - size - 1;
+
+                if (py::isinstance<py::ellipsis>(*it))
+                {
+                    break;
+                }
+                auto & slice_out = slices[ndim - size - 1];
+                const auto slice_in = (*it).cast<py::slice>();
+
+                copy_slice(slice_out, slice_in);
+            }
+        }
+    }
+
+    static void slice_syntax_check(pybind11::tuple const & tuple, size_t ndim)
+    {
+        namespace py = pybind11;
+
+        size_t ellipsis_cnt = 0;
+        size_t slice_cnt = 0;
+
+        for (auto it = tuple.begin(); it != tuple.end(); it++)
+        {
+            if (py::isinstance<py::ellipsis>(*it))
+            {
+                ellipsis_cnt += 1;
+            }
+            else if (py::isinstance<py::slice>(*it))
+            {
+                slice_cnt += 1;
+            }
+            else
+            {
+                throw std::runtime_error("unsupported operation.");
+            }
+        }
+
+        if (ellipsis_cnt + slice_cnt > ndim)
+        {
+            throw std::runtime_error("syntax error. dimensions mismatches");
+        }
+
+        if (ellipsis_cnt > 1)
+        {
+            throw std::runtime_error("syntax error. no more than one ellipsis.");
+        }
+    }
+
+    static void broadcast_array_using_slice(wrapped_type & arr_out,
+                                            std::vector<small_vector<int>> const & slices,
+                                            pybind11::array const & arr_in)
+    {
+        TypeBroadCast::check_shape(arr_out, slices, arr_in);
+
+        size_t nghost = 0;
```

Comment:
It'll be more readable:

```cpp
size_t const nghost = arr_out.nghost();
if (0 != nghost) { arr_out.set_nghost(0); }
TypeBroadCast ...;
if (0 != nghost) { arr_out.set_nghost(nghost); }
```

Reply:
There is a theory that constants or literals should be put as the first argument for comparison operation so that compiler will surely make a noise with a typo happens: `0 = nghost`.